### DISCUSSION TITLE
How to run custom scripts in a nuxt project?

### DISCUSSION THREAD

**Discussion by Nugine** [nuxt/nuxt#19662] [Questions]
I want to run a script to validate my content files before `nuxt generate`.
I put the parser logic in `utils/content.ts`. `server/api/content.ts` and `scripts/validate.ts` share the same logic.

How to run `scripts/validate.ts`?

```
.
├── content
│   └── index.md
├── server
│   └── api
│       └── content.ts
├── utils
│   └── content.ts
└── scripts
    └── validate.ts
```

**Comment by raven-dever** [comment]
My best guess is that you could just validate it inside a nuxt-module or plugin during a nuxt lifecycle that runs before the app starts or is built. As far as I know 'build:before' is the very first hook that get's called
https://nuxt.com/docs/api/advanced/hooks#nuxt-hooks-build-time
